internal/services/dynatrace/helper.go

```diff
@@ -36,6 +36,32 @@ func ExpandDynatraceUserInfo(input []UserInfo) *monitors.UserInfo {
 	})
 }
 
+func FlattenDynatraceEnvironmentProperties(input *monitors.DynatraceEnvironmentProperties) []EnvironmentProperties {
+	if input == nil {
+		return []EnvironmentProperties{}
+	}
+
+	environmentInfo := FlattenDynatraceEnvironmentInfo(input.EnvironmentInfo)
+
+	return []EnvironmentProperties{
+		{
+			EnvironmentInfo: environmentInfo,
+		},
+	}
+}
+
+func FlattenDynatraceEnvironmentInfo(input *monitors.EnvironmentInfo) []EnvironmentInfo {
+	if input == nil {
+		return []EnvironmentInfo{}
+	}
+
+	return []EnvironmentInfo{
+		{
+			EnvironmentId: pointer.From(input.EnvironmentId),
+		},
+	}
+}
```

Comment:
Couldn't these two functions be collapsed into a single function (e.g., `FlattenDynatraceEnvironmentProperties`)? I don't understand why we need the second `FlattenDynatraceEnvironmentInfo` function unless it is called from another resource? Also, should these be public functions vs. a private function (e.g., `FlattenDynatraceEnvironmentProperties` vs. `flattenDynatraceEnvironmentProperties`)?

```suggestion
func FlattenDynatraceEnvironmentProperties(input *monitors.DynatraceEnvironmentProperties) 
	result := []EnvironmentProperties{}

	if input == nil {
		return result 
	}

	if input.EnvironmentInfo != nil {
		result.EnvironmentInfo.EnvironmentId = pointer.From(input.EnvironmentInfo)
	}

	return result 
}
```

Reply:
@WodansSon , I'm not combining these 2 functions because there're other attributes under this `EnvironmentProperties` as well as `EnvironmentInfo`, the service said they will give us more details on how to configure them recently so I'll leave the functions uncollapsed.

Reply:
@jiaweitao001, I understand that, however, as more and more child `EnvironmentProperties` are exposed by the service team we would just need to inline those blocks into the one flatten function (e.g, `flattenDynatraceEnvironmentsProperties`), making the code much easier to read/understand and maintain moving forward for repo maintainers. If we end up with 12 flatten functions that get called from the parent flatten function it becomes exponentially more complicated to understand what the code is doing and review.